From 1bd8fc7d4b34f752a32709d277a897e5ad202d97 Mon Sep 17 00:00:00 2001
From: Tobias Heider <tobhe@users.noreply.github.com>
Date: Tue, 15 Aug 2023 18:06:48 +0200
Subject: [PATCH] privsep: fix strlcpy overflow in psp_ifname (#239)

When running our Ubuntu tests with libc6 and strlcpy overflow checks
enabled we found that the wrong size is passed to strlcpy resulting
in a crash because of an overflow.

Upstream-Status: Backport
[https://github.com/NetworkConfiguration/dhcpcd/commit/1bd8fc7d4b34f752a32709d277a897e5ad202d97]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 src/privsep.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/privsep.c b/src/privsep.c
index b11c0351..cfe54742 100644
--- a/src/privsep.c
+++ b/src/privsep.c
@@ -1200,7 +1200,7 @@ ps_newprocess(struct dhcpcd_ctx *ctx, struct ps_id *psid)
 #endif
 
 	if (!(ctx->options & DHCPCD_MANAGER))
-		strlcpy(psp->psp_ifname, ctx->ifv[0], sizeof(psp->psp_name));
+		strlcpy(psp->psp_ifname, ctx->ifv[0], sizeof(psp->psp_ifname));
 	TAILQ_INSERT_TAIL(&ctx->ps_processes, psp, next);
 	return psp;
 }
-- 
2.25.1

